src/rastervision/common/data/generators.py

```diff
@@ -66,11 +67,22 @@ def __init__(self, active_input_inds, train_ratio, cross_validation):
         if self.cross_validation is not None:
             self.process_cross_validation()
 
-        gen = self.make_split_generator(
-            TRAIN, target_size=(10, 10), batch_size=100, shuffle=True,
-            augment=False, normalize=False, only_xy=False)
-        batch = next(gen)
-        self.normalize_params = get_channel_stats(batch.all_x)
+        """
+        if a specific dataset's normalized parameters have already been
+        calculated, then load that file. Otherwise, calculate with small batch.
+        """
+        const_path = join(self.dataset_path, self.name+'_norm_constants.json')
+        if exists(const_path):
+            obj_text = codecs.open(const_path, 'r', encoding='utf-8').read()
```

Comment:
Is there a reason for using the codecs package? If not, I'd prefer loading json according to prior examples in the repo such as https://github.com/azavea/raster-vision/blob/076b5f9a3b381a3c256916644a6f61bcee8c1150/src/rastervision/run.py#L32-L33